projects/observability/src/pages/explorer/explorer.component.ts

```diff
@@ -227,12 +227,21 @@ export class ExplorerComponent {
 
     const currentQuery = { scopeQueryParam: currentScope, filters: cleanedFilters };
 
-    if (this.explorerService.isDuplicateQuery(currentQuery, this.savedQueries)) {
+    if (
+      this.explorerService.isDuplicateQuery(
+        currentQuery,
+        this.savedQueries.map(({ name, ...rest }) => rest)
+      )
+    ) {
       this.notificationService.createInfoToast('This query is saved already!');
     } else {
-      const newSavedQueries = [...this.savedQueries, currentQuery];
-      this.preferenceService.set('savedQueries', newSavedQueries);
-      this.notificationService.createSuccessToast('Query Saved Successfully!');
+      const queryName = prompt('Please enter a name for this query', 'My query');
```

Comment:
Should we use a prompt or a modal here? I think prompt should work for the MVP maybe we can add this as an enhancement in future // @jaywalker21

Reply:
Yes, I was thinking about that too. But I figured changing the prompt to modal with a custom input wouldn't really change the UX. But agree, it's good to have in the long run for a better UI. Low priority though. 

Captured the requirement here:
https://razorpay.atlassian.net/browse/OBSV-642